### ISSUE TITLE
🚀 Feature Request: Allow user-uploaded workers to register as queue consumers

### ISSUE THREAD

**Issue by jasonbarry** [cloudflare/workers-sdk#6758]
### Describe the solution

There's [no indication](https://developers.cloudflare.com/cloudflare-for-platforms/workers-for-platforms/platform/limits/) that user-uploaded scripts cannot be queue consumers, however in practice that seems to be the case (please correct me if that's not true or if there's an undocumented workaround!). 

The [Create Queue Consumer](https://developers.cloudflare.com/api/operations/queue-v2-create-queue-consumer) endpoint doesn't work if you pass in the name of a user-uploaded script. That script lives within the Workers for Platforms namespace, so it makes sense that the script cannot be found. But there doesn't exist an endpoint [under the Workers for Platforms section](https://developers.cloudflare.com/api/operations/namespace-worker-list) that lets user workers register as queue consumers. 

Similarly, there doesn't exist a [binding type](https://developers.cloudflare.com/workers/configuration/multipart-upload-metadata/#bindings) in the Upload API that binds a user worker to a queue consumer -- the `queue` type only binds workers to queue producers. 

If user-uploaded scripts could be bound to a queue consumer, it would unlock some pretty powerful workflows! As it stands, the `queue()` method defined in a user-uploaded worker's default export is currently unreachable, as far as I understand it. Thanks for your time and consideration!

**Comment by penalosa** [comment]
To clarify, is this with Workers for Platforms? This should work with normal workers, as far as I know

**Comment by jasonbarry** [comment]
@penalosa Yes, this is for WfP. I've since worked around this by putting the queue consumer in the dispatch worker, and then forwarding all requests in a POST call to the relevant user workers. 

My two cents for the resolution of this issue is to add a paragraph in https://developers.cloudflare.com/cloudflare-for-platforms/workers-for-platforms/platform/limits/ that states that the `queue()` handler (and potentially other non-`fetch` [handlers](https://developers.cloudflare.com/workers/runtime-apis/handlers/)?) are not reachable when deployed as a user worker.
